woocommerce/woocommerce
# Add vendor bundling of blocks frontend bundle to improve cart/checkout performance

### Changes proposed in this Pull Request:

Recently I discovered while reviewing the cart and checkout webpack builds that there was a relatively low-hanging opportunity for optimization.

Specifically we don't do any vendor bundling in the front-end webpack bundle. There are so many entrypoints in this bundle, so even one small dependency being duplicated across all of them, drastically impacts the payload size for cart and checkout. This is especially true because checkout for example, has many inner blocks that are all entrypoints which will all end up duplicating any vendor dependency.

Furthermore, the `blocks-checkout` and `blocks-components` scripts that were being built in a separate bundle were not able to make use of vendor bundling because most of their dependencies were duplicated from the front-end bundle. 

So to solve this, I moved those bundles across to the front-end bundle and turned on vendor bundling of node_modules there. On a production build, in testing, this reduced the payload of the cart page by `11%` and the checkout page by `17.9%`. 

Closes https://github.com/woocommerce/woocommerce/issues/45680



### How to test the changes in this Pull Request:

I am leaning in on the e2e tests here to see that this still works. For tester though you should do a smoke test of:

1. Shop page and functionality on frontend
2. Cart page and functionality on frontend
3. Checkout page and functionality on frontend

And also you should test all of the above in the editor as well. Try changing block settings, add / remove the blocks and ensure that no JavaScript exceptions have been thrown.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Introduce vendor bundling to the blocks cart and checkout pages to improve performance.

#### Comment 

</details>
